src/posit/connect/external/databricks.py

```diff
@@ -8,38 +8,32 @@
 https://github.com/databricks/databricks-sdk-py#interface-stability
 """
 
-import abc
+from __future__ import annotations
 
-import requests
-from typing_extensions import Callable, Dict, Optional
+import logging
 
-from .._utils import is_local
-from ..client import Client
-from ..oauth import Credentials
-
-POSIT_OAUTH_INTEGRATION_AUTH_TYPE = "posit-oauth-integration"
-POSIT_LOCAL_CLIENT_CREDENTIALS_AUTH_TYPE = "posit-local-client-credentials"
+from typing_extensions import TYPE_CHECKING, Dict, Optional
 
-# The Databricks SDK CredentialsProvider == Databricks SQL HeaderFactory
-CredentialsProvider = Callable[[], Dict[str, str]]
+from .._utils import is_connect, is_workbench
+from ..client import Client
 
+try:
+    from databricks.sdk.core import Config, DefaultCredentials
+    from databricks.sdk.credentials_provider import (
+        CredentialsProvider,
+        CredentialsStrategy,
+    )
+except ImportError as e:
+    raise ImportError("The 'databricks-sdk' package is required to use this module.") from e
 
-class CredentialsStrategy(abc.ABC):
-    """Maintain compatibility with the Databricks SQL/SDK client libraries.
+if TYPE_CHECKING:
+    from ..oauth import Credentials
 
-    See Also
-    --------
-    * https://github.com/databricks/databricks-sql-python/blob/v3.3.0/src/databricks/sql/auth/authenticators.py#L19-L33
-    * https://github.com/databricks/databricks-sdk-py/blob/v0.29.0/databricks/sdk/credentials_provider.py#L44-L54
-    """
 
-    @abc.abstractmethod
-    def auth_type(self) -> str:
-        raise NotImplementedError
+POSIT_OAUTH_INTEGRATION_AUTH_TYPE = "posit-oauth-integration"
+POSIT_WORKBENCH_AUTH_TYPE = "posit-workbench"
 
-    @abc.abstractmethod
-    def __call__(self, *args, **kwargs) -> CredentialsProvider:
-        raise NotImplementedError
+logger = logging.getLogger("posit.sdk")
```

Comment:
I'm pretty sure we can use `logger = logging.getLogger(__name__)` here. Which will be equivalent to `getLogger("posit.sdk.external.databricks")`.

https://docs.python.org/3/howto/logging.html#advanced-logging-tutorial